src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IQueueItem.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+#nullable enable
+
+public interface IQueueItem<RequestContextType>
```

Comment:
Given that `QueueItem` is in this PR too, is this needed? Is this just hedging? There is a lot to this that is implementation details of the `RequestExecutionQueue`, so trying to design this interface nicely will be difficult. Can we avoid this and just use QueueItem directly, and not expose it at all?

Reply:
`IQueueItem` is the base-type that we can use for the Queue since `QueueItem<RequestContextType, RequestType, ResponseType>` wouldn't allow QueueItems from different handlers onto the same queue. You're correct that it doesn't need to be public though.

Reply:
Oh, this actually DOES need to be public so that it can be a part of the public interface `IRequestContextFactory.CreateRequestContextAsync`

Reply:
imho then IQueueItem should just essentially be a marker interface, or have a very limited surface area.  And leave 90% to the implementation